NewsFeed/Common/Views/ScrollableSegmentedControl.swift

```diff
@@ -0,0 +1,197 @@
+//
+//  ScrollableSegmentedControl.swift
+//  ScrollableSegmentedControl
+//
+//  Created by MArko Satlan on 22/07/2018.
+//  Copyright © 2018 MArko Satlan. All rights reserved.
+//
+
+import UIKit
+
+class ScrollableSegmentedControl: UIControl {
+
+// MARK: - Properties
+    // Open
+    open var font = UIFont(name: "Georgia", size: 15) {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+    open var buttonSpacing: CGFloat = 15 {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+    open var buttonMargin: CGFloat = 15 {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+    open var selectedSegmentIndex: Int = -1 {
+        didSet {
+            highlightSelectedSegment()
+            setScrollOffset()
+      
+            // deselect previous button if different button is tapped
+            if oldValue != selectedSegmentIndex {
+                if let previousButton = (buttonsArray.filter{$0.tag == oldValue}.first) {
+                    previousButton.isSelected = false
+                    UIButton.animate(withDuration: 0.3, animations: {
+                        previousButton.transform = CGAffineTransform.identity
+                    })
+                }
+            }
+        }
+    }
+    
+    // Private
+    private var buttonTitles: [String] = []
+    private var buttonsArray: [UIButton] = []
+    private let scrollView = UIScrollView()
+    private let separatorLine = UIView()
+    
+// MARK: - Methods
+    // Init
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        
+        setUp()
+    }
+    
+    public init(items: [String]) {
+        self.buttonTitles = items
+        
+        super.init(frame: CGRect.zero)
+        
+        setUp()
+        createButtons()
+    }
+    
+    public func setItems(_ items: [String]) {
+        self.buttonTitles = items
+        createButtons()
+        setNeedsLayout()
+    }
+    
+    // Layout subviews - set frames
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        
+        scrollView.frame = bounds
+        
+        separatorLine.frame = CGRect(x: 0,
+                                     y: frame.size.height - 1,
+                                     width: frame.size.width,
+                                     height: 1)
+        
+        updateUI()
+        highlightSelectedSegment()
+        setScrollOffset()
+    }
+    
+    // Action methods
+    @objc private func buttonTapped(_ sender: UIButton) {
+        if let previousButton = (buttonsArray.filter{$0.tag == selectedSegmentIndex}.first) {
+            if previousButton.tag != sender.tag {
+                sendActions(for: .valueChanged)
```

Comment:
Ooops. This looks like a potential bug. You are sending an event here, but new `selectedSegmentIndex` is set afterwards, at line `104`. Was the control still working properly after this change?

Anyways, this one's on me, as that was my (bad) advice to handle previous button and then current. It seems that you'll need to set `selectedSegmentIndex = sender.tag` before `sendActions(for: .valueChanged)`.

Reply:
It's a bug either way (whether selectedSegmentIndex = sender.tag is before or after previous button) - can be reproduced in a case when there is no segment selected (-1)